### PR TITLE
Test fixes

### PR DESCRIPTION
**Pull request by zooba**
Fixes a number of deadlocks and assertions within tests (and potentially within the product too)

### CODE REVIEW

**Comment by crwilcox** [comment]
:+1:

**Comment by int19h** [comment]
:+1:

File: Python/Product/PythonTools/PythonTools/Intellisense/TaskProvider.cs [hunk 1]
```diff
@@ -461,79 +449,76 @@ abstract class TaskProvider : IVsTaskProvider, IDisposable {
...
-                            }
                         }
+                        msg = _workerQueue.Dequeue();
                     }
 
-                    if (cts.IsCancellationRequested) {
-                        break;
+                    if (msg is WorkerMessage.FlushMessage) {
+                        // Keep flush messages until we've exited the loop
+                        flushMessages.Enqueue(msg);
```
Comment (gilbertw) [inline Python/Product/PythonTools/PythonTools/Intellisense/TaskProvider.cs]: I don't see flushMessages getting cleaned up.  Is it ok to just let that build infinitely?

**Comment by gilbertw** [comment]
:+1:

**Pull request merged**